decidim-participatory_processes/app/controllers/decidim/participatory_processes/widgets_controller.rb

```diff
@@ -5,13 +5,19 @@ module ParticipatoryProcesses
     class WidgetsController < Decidim::WidgetsController
       helper Decidim::SanitizeHelper
 
+      def show
+        enforce_permission_to :embed, :participatory_space, current_participatory_space: model
+
+        super
+      end
+
       private
 
       def model
         return unless params[:participatory_process_slug]
 
-        @model ||= ParticipatoryProcess.where(slug: params[:participatory_process_slug]).or(
-          ParticipatoryProcess.where(id: params[:participatory_process_slug])
+        @model ||= ParticipatoryProcess.where(organization: current_organization).published.where(slug: params[:participatory_process_slug]).or(
+          ParticipatoryProcess.where(organization: current_organization).published.where(id: params[:participatory_process_slug])
```

Comment:
To avoid redirect, i would use. 
```suggestion
        @model ||= ParticipatoryProcess.where(organization: current_organization).public_spaces.where(slug: params[:participatory_process_slug]).or(
          ParticipatoryProcess.where(organization: current_organization).public_spaces.where(id: params[:participatory_process_slug])
```